src/controllers/clientController.js

```diff
@@ -137,3 +137,41 @@ exports.getChefs = async (req, res) => {
       .json({ error: `Error while retrieving chefs: ${error.message}` });
   }
 };
+
+exports.dishesRatings = async (req, res) => {
+  const token = req.cookies.jwt;
+  const decodedToken = decodeJwtToken(token);
+
+  const { id } = req.params;
+  const { rating } = req.body;
+
+  try {
+    const dish = await Dish.findById(id);
+
+    const existingRatingIndex = dish.ratings.findIndex((r) =>
+      r.userId.equals(decodedToken.userId)
+    );
+
+    if (existingRatingIndex !== -1) {
+      dish.ratings[existingRatingIndex].rating = rating;
+    } else {
+      const newRating = { userId: decodedToken.userId, rating };
+      dish.ratings.push(newRating);
+    }
```

Comment:
I think we agreed on a client can only rate the dish once, so we might change this to 
if (existingRatingIndex !== -1) --> error message: You can not rate the same dish again
else.. --> the same